### PR TITLE
Refactor app-config utils

### PR DESCRIPTION
**Pull request by jahow**
This is an attempt to simplify the code used for parsing the app configuration file and make it more readable. Hopefully this will improve the app-config code maintanability.

### CODE REVIEW

File: libs/util/app-config/src/lib/parse-utils.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+): Record<string, unknown>[] | null {
+  const sections = fullConfigObj[sectionName]
+  if (!sections) {
+    return []
+  }
+  if (!Array.isArray(sections)) {
+    outErrors.push(`Expected [[${sectionName}]] to be an array of sections.`)
+    return null
+  }
+  const result = sections.map((section) =>
```
Comment (tkohr) [inline libs/util/app-config/src/lib/parse-utils.ts:109]: Nice, thanks for adding this and checking the layer properties as well!

File: libs/util/app-config/src/lib/app-config.ts [hunk 2]
```diff
@@ -95,150 +84,99 @@ export function loadAppConfig() {
...
-        BACKGROUND_COLOR: theme.background_color,
-        HEADER_BACKGROUND: theme.header_background,
-        TITLE_FONT: theme.title_font,
-        MAIN_FONT: theme.main_font,
-        FONTS_STYLESHEET_URL: theme.fonts_stylesheet_url,
       }
-      customTranslations = translations || {}
 
       appConfigLoaded = true
     })
```
Comment (tkohr) [inline libs/util/app-config/src/lib/app-config.ts:182]: Do we actually still need the `_reset()` function below ?

**Review by tkohr — APPROVED** [review]
LGTM, thanks @jahow !

File: libs/util/app-config/src/lib/app-config.ts (hunk 2 above)
Comment (jahow) [reply libs/util/app-config/src/lib/app-config.ts:182]: I think so, for tests in `app-config.spec`

**Pull request merged**